### PR TITLE
Add tests for various modules, fix @ritzvik 's issue

### LINKED ISSUE #378: Increase code coverage across different modules (easy ones)
🐞 **Problem**
The goal is 100% code coverage. It's not a very realistic goal because sometimes we lack the knowledge to do so. But I am listing some of the parts, where coverage can be increased easily. 

🎯 **Goal**

Try to target these files. The lines untested would be marked red [here](https://codecov.io/gh/einsteinpy/einsteinpy/tree/master/src/einsteinpy).
  - [x] `coordinates/core.py` (#394)
  - [x] `ijit.py` (#397)
  - [x] `symbolic/vector.py`
  - [x] `symbolic/tensor.py`
  - [x] `plotting/hypersurface/core.py`
  - [x] `plotting/rays/shadow.py`
  - [x] `rays/shadow.py`

💡 **Possible solutions**

  - [What is coverage](https://confluence.atlassian.com/clover/about-code-coverage-71599496.html)
  - tests are present in `src\einsteinpy\tests\test_<module name>\test_<file name>`

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

@shreyasbapat if possible, add some more!

### PR DESCRIPTION
**Pull request by shreyasbapat**
Fixes #378

### CODE REVIEW

**Comment by shreyasbapat** [comment]
Alright. this is still WIP

**Comment by shreyasbapat** [comment]
Halting this for meeting!

**Comment by ritzvik** [comment]
You want to drop the brackets for perihelion and I want lazy computation. That’s right?

**Comment by ritzvik** [comment]
I have an idea

**Comment by ritzvik** [comment]
Create an example class and make various examples as its properties🙃

**Comment by shreyasbapat** [comment]
I had the class idea. But :/

It's a pain for users to understand how that is working. Do you have a
design in mind?

**Comment by ritzvik** [comment]
> I had the class idea. But :/ It's a pain for users to understand how that is working. Do you have a design in mind?
> […](#)

Something like this
```
from einsteinpy import examples 
examples.perihelion
```

**Comment by shreyasbapat** [comment]
> > I had the class idea. But :/ It's a pain for users to understand how that is working. Do you have a design in mind?
> > […](#)
> 
> Something like this
> 
> ```
> from einsteinpy import examples 
> examples.perihelion
> ```

Which one do you think is more user-friendly. The one in this PR or the one you are proposing. I am unable to decide.

**Comment by ritzvik** [comment]
The first one seems more user friendly. The one in the PR.

Aside than this, If you think my point is valid, I can see after the bodies module.

File: src/einsteinpy/tests/test_plotting/test_rays/test_shadow.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    assert isinstance(cl.shadow, Shadow)
+    assert cl.is_intensity_plot
+
+
+@mock.patch("einsteinpy.plotting.rays.shadow.plt.plot")
+def test_plot_calls_plt_plot(mock_patch, dummy_data):
+    shadow = dummy_data
+    cl = ShadowPlotter(shadow=shadow)
+    cl.plot()
+    mock_patch.assert_called_with(cl.shadow.fb1, cl.shadow.intensity, "r")
```
Comment (ritzvik) [inline src/einsteinpy/tests/test_plotting/test_rays/test_shadow.py]:
```suggestion
    expected = [
        mock.call(cl.shadow.fb1, cl.shadow.intensity, "r"),
        mock.call(cl.shadow.fb2, cl.shadow.intensity, "r"),
    ]
    assert mock_patch.call_args_list == expected
```

Thank me later :p

Comment (shreyasbapat) [reply src/einsteinpy/tests/test_plotting/test_rays/test_shadow.py]: Yeah I forgot there were two calls.

**Comment by ritzvik** [comment]
Any progress here??

**Comment by ritzvik** [comment]
> Any progress here??

😖

**Comment by shreyasbapat** [comment]
Things are moving slowly at my end. Have patience.

**Comment by shreyasbapat** [comment]
Rebased

**Comment by shreyasbapat** [comment]
Hmmmmmm. Github actions is really slow

**Comment by ritzvik** [comment]
> Hmmmmmm. Github actions is really slow

It's running for 30 mins :D

**Comment by ritzvik** [comment]
It got stuck. I reran the job and it failed :(
Guess, you can merge it and see what happens

**Comment by shreyasbapat** [comment]
Gooooooooood

**Pull request merged**